### PR TITLE
fix panic when close miner

### LINKED ISSUE #312: Panic while closing miner
```
goroutine 936 [running]:
reflect.chansend(0xc00028e720, 0xc00035b420, 0x100, 0x1998e21)
        /usr/lib/go/src/runtime/chan.go:665 +0x4b
reflect.Value.send(0x1789d20, 0xc00028e720, 0x12, 0x17764a0, 0xc00035b420, 0x197, 0x0, 0x0)
        /usr/lib/go/src/reflect/value.go:1524 +0x118
reflect.Value.Send(0x1789d20, 0xc00028e720, 0x12, 0x17764a0, 0xc00035b420, 0x197)
        /usr/lib/go/src/reflect/value.go:1506 +0x90
github.com/filecoin-project/go-lotus/lib/jsonrpc.(*client).makeOutChan.func1.1.1(0xc00024a570, 0xc000354328, 0x1789d20, 0xc00028e720, 0x12)
        /root/go-lotus/lib/jsonrpc/client.go:177 +0xb0
created by github.com/filecoin-project/go-lotus/lib/jsonrpc.(*client).makeOutChan.func1.1
        /root/go-lotus/lib/jsonrpc/client.go:171 +0x47c
```

### PR DESCRIPTION
**Pull request by deaswang**
fix #312

### CODE REVIEW

File: lib/jsonrpc/client.go [hunk 1]
```diff
@@ -173,13 +175,28 @@ func (c *client) makeOutChan(ctx context.Context, ftyp reflect.Type, valOut int)
...
+						{
+							Dir:  reflect.SelectSend,
+							Chan: ch,
+							Send: front.Value.(reflect.Value).Elem(),
+						},
+					}
+
+					switch chosen, _, _ := reflect.Select(cases); chosen {
+					case 0:
+						bufLk.Lock()
```
Comment (magik6k) [inline lib/jsonrpc/client.go]: I'd move this before the switch (with the select), this way we only call Lock once

Comment (deaswang) [reply lib/jsonrpc/client.go]: but it should lock after channel send.

Comment (magik6k) [reply lib/jsonrpc/client.go]: Just put `chosen, _, _ := reflect.Select(cases)` before the loop as well

Comment (deaswang) [reply lib/jsonrpc/client.go]: then how to get front value?

Comment (deaswang) [reply lib/jsonrpc/client.go]: oh.I Got it.

**Pull request merged**